fiber/src/core.ml

```diff
@@ -384,27 +384,27 @@ let rec repeat_while : 'a. f:('a -> 'a option t) -> init:'a -> unit t =
   | None -> return ()
   | Some init -> repeat_while ~f ~init
 
-let collect_errors f =
+let collect_errors_appendable_list f =
   let module Exns = Monoid.Appendable_list (Exn_with_backtrace) in
-  let+ res =
-    map_reduce_errors
-      (module Exns)
-      f
-      ~on_error:(fun e -> return (Appendable_list.singleton e))
-  in
-  match res with
-  | Ok x -> Ok x
+  map_reduce_errors
+    (module Exns)
```

Comment:
I'm never sure if the compiler can float this to the top level to avoid doing some work on every call. How about manually pulling this out?

```
module Exn_appendable_list = Monoid.Appendable_list (Exn_with_backtrace)

let collect_errors_appendable_list f =
  map_reduce_errors
    (module Exn_appendable_list)
```

Reply:
> I'm never sure if the compiler can float this to the top level

I guess it can't because the functor can be doing some side-effects, e.g. incrementing a counter.